src/recipe.cpp

```diff
@@ -729,32 +738,40 @@ std::set<proficiency_id> recipe::assist_proficiencies() const
     return ret;
 }
 
-float recipe::proficiency_time_maluses( const Character &guy ) const
+float recipe::proficiency_time_maluses( const Character &c ) const
 {
-    float total_malus = 1.0f;
+    double total_malus = 1.0;
     for( const recipe_proficiency &prof : proficiencies ) {
-        if( !guy.has_proficiency( prof.id ) &&
-            !helpers_have_proficiencies( guy, prof.id ) && prof.time_multiplier > 1.0f ) {
-            float malus = 1.0f + ( prof.time_multiplier - 1.0f ) *
-                          ( 1.0f - guy.crafting_inventory().get_book_proficiency_bonuses().time_factor( prof.id ) );
-            total_malus *= malus;
+        if( !c.has_proficiency( prof.id ) &&
+            !helpers_have_proficiencies( c, prof.id ) && prof.time_multiplier > 1.0f ) {
+            double malus = prof.time_multiplier - 1.0;
+            malus *= 1.0 - c.crafting_inventory().get_book_proficiency_bonuses().time_factor( prof.id );
+            double pl = get_aided_proficiency_level( c, prof.id );
+            // Sigmoid function that mitigates 100% of the time malus as pl approaches 1.0
+            // but has little effect at pl < 0.5. See #49198
+            malus *= 1.0 - std::pow( 0.5 + 0.5 * std::sin( pl * M_PI - M_PI / 2 ), 2 );
+            total_malus *= 1.0 + malus;
```

Comment:
Would it make sense to keep this formula contained to proficiency_set or a character function? e.g. would we want to use this formula somewhere else?

Reply:
This is reused, but only in this file. Please have another look post-refactor.

Reply:
I'm asking more if we would want to use this function anywhere else. If we're just going to use it for crafting, it can live here, but if we're going to possibly use it more places, I think we could plan ahead.

Reply:
Well, I can't think of any reason why we'd want the penalty for a single proficiency other than for display purposes. What matters for simulation purposes is the total penalty taking into account all proficiencies, which is already exported as a public function.